### PR TITLE
Jetpack DNA: Move Sync Comments Module from Legacy to psr-4

### PR DESCRIPTION
**Pull request by gititon**
This PR moves the sync comments module from legacy to PSR-4.

No new tests should be needed because this is a refactor, so existing tests should catch any regressions.

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "packages": [
         {
             "name": "automattic/jetpack-asset-tools",
-            "version": "dev-master",
```
Comment (lezama) [inline composer.lock]: we shouldn't push this changes to master.

File: packages/sync/src/modules/Comments.php [hunk 2]
```diff
@@ -178,7 +180,7 @@ function filter_meta( $args ) {
 
 	public function expand_comment_ids( $args ) {
 		list( $comment_ids, $previous_interval_end ) = $args;
-		$comments    = get_comments(
+		$comments                                    = get_comments(
```
Comment (lezama) [inline packages/sync/src/modules/Comments.php]: oops

File: tests/php/sync/test_class.jetpack-sync-comments.php [hunk 3]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+use Automattic\Jetpack\Sync\Modules\Comments;
```
Comment (lezama) [inline tests/php/sync/test_class.jetpack-sync-comments.php]: we don't seem to be using this class in this file

File: packages/sync/src/modules/Comments.php (hunk 2 above)
Comment (tyxla) [reply packages/sync/src/modules/Comments.php]: This seems ok - see the statement above.

**Review by tyxla — COMMENTED** [review]
Looking good, just needs the extra `use` call removed.

File: packages/sync/src/modules/Comments.php (hunk 2 above)
Comment (lezama) [reply packages/sync/src/modules/Comments.php]: true!

**Review by lezama — APPROVED** [review]

**Pull request merged**